src/converter/FlameGraph.java

```diff
@@ -173,7 +177,9 @@ private String stripSuffix(String title) {
     }
 
     private int frameType(String title) {
-        if (title.endsWith("_[j]")) {
+        if (highlightStacksPattern.matcher(title).matches()) {
+            return 5;
```

Comment:
The problem is, such highlighting discards the information about the original frame type.
I'd rather not add a new frame type here, but do highlighting on the client side. The easiest way will be to initialize `pattern` variable in the generated JavaScript code.

Reply:
To the best of my reading, using `pattern` would prevent the ability of using the search box in the generated page. The frame type only seems to be used to colorize the stack, which is the intent of this use case. Let me know if I have misinterpreted how this might work in the client-side.

The goal was to provide a way for a user to define a custom frame type that matches their needs. For example, they may want a classification of frames for "application domain calls" which all match a specific package name. I know that matches the goal of my organization, to separate the frame type for the code our organization owns in the graph from those outside of our organization, while still being able to use the search field to search for other interesting elements in the graph.

Reply:
It could be better to put your pattern(s) with their colors to generated JS code:
```
    "\tfunction f(level, left, width, type, title) {\n"
    "\t\tlevels[level].push({left: left, width: width, color: getColor(palette[type]), title: title});\n"
    "\t}\n"
```
Here we can easily match any number of patterns against the title and choose appropriate colors without touching original type.

Reply:
Okay, I've rewritten it to just change the palette color, feeding the pattern into the javascript function and applying the extra color style if anything matches.